src/parser/statement.js

```diff
@@ -303,31 +304,39 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop. The lookahead token must be a
+    // var name. If it isn't, someone is using one of the contextual keywords as a var name.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val")) &&
+      (this.lookahead().type === tt.name)
```

Comment:
I haven't looked at whether this is necessary or not (it might be) but just a heads-up that lookaheads are pretty unfortunate in terms of perf – they copy the entire state object, twice. 

There's a good chance that we can start a `parseEnhancedForIn` with just the `isContextual` checks and, if nothing follows them, return an ordinary for-in. 

...actually, come to think of it, since we're making `for x in y` illegal - instead requiring `for const x in y` or `for now x in y` – we don't need the lookahead at all, I think.